### PR TITLE
chore(addon): #2204 Remove url-parse and replace with URL

### PR DESCRIPTION
**Pull request by k88hudson**
- Removes `url-parse` and replaces it with `URL`
- Adds a `hostname` property in `PreviewProvider`
- Consolidates cache-key-related functions for deduping
- Removes `siteMetadataSelectors` and refactors `prettyUrl` to use the new`.hostname` property

### CODE REVIEW

File: addon/PreviewProvider.js [hunk 1]
```diff
@@ -46,6 +46,13 @@ function PreviewProvider(tabTracker, metadataStore, store, options = {}) {
   this.init();
 }
 
+function createCacheKey(spec) {
+  let url = new URL(spec);
+  let key = url.host.replace(/www\.?/, "");
+  key = key + url.pathname + (url.search || "");
+  return key.toString();
+}
```
Comment (pdehaan) [inline addon/PreviewProvider.js]: This seems like a copy/pasta of common/create-cache-key.js below.

File: addon/PreviewProvider.js [hunk 2]
```diff
@@ -46,6 +46,13 @@ function PreviewProvider(tabTracker, metadataStore, store, options = {}) {
   this.init();
 }
 
+function createCacheKey(spec) {
```
Comment (sarracini) [inline addon/PreviewProvider.js]: Since we removed create-cache-key.js, maybe revert this change to pulling out createCacheKey?

**Comment by as-pine-proxy** [comment]
[Treeherder: 3ed7f7b0945467c2f886cc728812833f2f6e9919](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=3ed7f7b0945467c2f886cc728812833f2f6e9919)
